model/point_model.py

```diff
@@ -113,6 +119,52 @@ def set_pulse_shape(
         """
         self._pulse_generator = pulse_generator
 
+    def add_noise(
+        self,
+        noise_to_signal_ratio: float,
+        seed: Union[None, int] = None,
+        noise_type: str = "additive",
+    ):
+        """
+        Adds noise to the realization.
+        Parameters
+        ----------
+        noise_to_signal_ratio: float, defined as X_rms/S_rms where X is noise and S is signal.
+        seed: None or int, seed for the noise generator
+        noise_type: str
+            "additive": additive noise
+            "dynamic": dynamic noise (only applicable for constant duration times)
+            "both": both additive and dynamic noise
+        """
+        assert noise_type in {"additive", "dynamic", "both"}
+        assert seed is None or isinstance(seed, int)
+        assert noise_to_signal_ratio >= 0
```

Comment:
> I would first think that the noise realization should be made when we realize the process, i.e. in the make_realization function.

Moving the realization of the noise to `make_realization` would make the structure clearer, I agree. I'll take a look at that tomorrow. 

> Otherwise, if we make several realizations of the process with the same model, all of them will use the same noise realization, isn't it?

Yes, they would all have the same noise realization. I haven't considered making several realizations of the same model. Moving the noise generation to `make_realization` should fix this.